integrations/azure-devops/azure_devops/http/http_client.py

```diff
@@ -0,0 +1,150 @@
+import httpx
+from httpx import Timeout, BasicAuth, Response
+from typing import Any, AsyncGenerator, Optional
+from port_ocean.utils import http_async_client
+from loguru import logger
+
+PAGE_SIZE = 50
+REQUEST_TIMEOUT_SECONDS = 30
+CONTINUATION_TOKEN_HEADER = "x-ms-continuationtoken"
+ALLOW_REDIRECTS = True
+
+
+class HTTPBaseClient:
+    def __init__(self, personal_access_token: str) -> None:
+        self._async_client = http_async_client
+        # Username isn't required in basic auth to Azure Devops
+        self._auth = BasicAuth("", personal_access_token)
+        self._timeout = Timeout(REQUEST_TIMEOUT_SECONDS)
+        self._follow_redirects = ALLOW_REDIRECTS
+
+    @staticmethod
+    def _parse_response_values(response: Response) -> list[dict[Any, Any]]:
+        return response.json()["value"]
+
+    def send_sync_get_request(
+        self, url: str, params: Optional[dict[str, Any]] = None
+    ) -> Response:
+        try:
+            response = httpx.get(
+                url,
+                params=params,
+                auth=self._auth,
+                timeout=self._timeout,
+                follow_redirects=self._follow_redirects,
+            )
```

Comment:
why directly using httpx?

Reply:
This is the method to send synchronous request from httpx, Couldn't do it from the async client.

Reply:
Lets talk about this face to face